mllib-dal/src/main/scala/com/intel/oap/mllib/OneDAL.scala

```diff
@@ -264,11 +264,13 @@ object OneDAL {
         columnIndices(indexValues) = column + 1
 
         if (row > curRow) {
```

Comment:
could you try to remove curRow, I think it's possible to remove curRow by just using row to scan the table.